flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -337,6 +342,11 @@ private Module createModule(Map<String, String> moduleProperties, ClassLoader cl
 		return factory.createModule(moduleProperties);
 	}
 
+	private void createAndRegisterCatalog(String name, CatalogEntry entry) {
+		Catalog catalog = createCatalog(name, entry.asMap(), Thread.currentThread().getContextClassLoader());
```

Comment:
Why `Thread.currentThread().getContextClassLoader()` instead of class member `classLoader`?
The only thing you need is `wrapClassLoader`.